### PR TITLE
Maps feature

### PR DESCRIPTION
**Pull request by Iamogeee**
This pull request introduces the integration of Google Maps into the business details page. The feature aims to enhance user experience by providing an interactive map displaying the exact location of the business.

**Changes Implemented**

Google Maps Integration:

- Utilized @vis.gl/react-google-maps package to incorporate Google Maps API.
- Added an interactive map to the business details page, displaying the business's location.

Map Marker and InfoWindow:

- Added a marker to the map indicating the business's location.

Project Demo: 
<div>
    <a href="https://www.loom.com/share/a34d85c2643c4c80a6f28fe9e95e87d5">
      <p>Loom Message - 16 July 2024 - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/a34d85c2643c4c80a6f28fe9e95e87d5">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/Components/BusinessDetails/BusinessDetails.jsx [hunk 1]
```diff
@@ -89,8 +119,36 @@ const BusinessDetails = () => {
...
           )}
           <div className="business-main-content">
-            <div className="business-section business-overview">
-              <BusinessOverview overview={business.overview} />
+            <div className="business-section business-map">
+              <APIProvider apiKey={apiKey}>
+                <Map
+                  defaultZoom={20}
+                  defaultCenter={parseLocation(business.location)}
+                  mapId={"218a557688af104"}
```
Comment (jose-esquivel) [inline frontend/src/Components/BusinessDetails/BusinessDetails.jsx:127]: This looks like it should be added to your .env file.

File: frontend/src/Components/SavedBusinesses/SavedBusinesses.css
Comment (jose-esquivel) [inline frontend/src/Components/SavedBusinesses/SavedBusinesses.css:1]: No need to commit empty files, please delete these before committing if you didn't add any code.

Comment (Iamogeee) [reply frontend/src/Components/SavedBusinesses/SavedBusinesses.css:1]: Left it there just incase I needed it but I'll take it out now

File: frontend/src/Components/BusinessDetails/BusinessDetails.jsx (hunk 1 above)
Comment (Iamogeee) [reply frontend/src/Components/BusinessDetails/BusinessDetails.jsx:127]: Missed that! Thanks, will do

**Comment by sinap-fb** [comment]
BTW Is it poosible to merge Rec-system-live-version-3 into this branch? I think this would make reviewing much easier

**Comment by Iamogeee** [comment]
> BTW Is it poosible to merge Rec-system-live-version-3 into this branch? I think this would make reviewing much easier

@sinap-fb Hey! Jose advised to merge into the main branch and then merge this one

**Pull request merged**